### PR TITLE
Initial commit moving frontend ui tests.

### PR DESCRIPTION
**Pull request by jrbenny35**
Works on #8910 and #9017

### CODE REVIEW

**Comment by jrbenny35** [comment]
These tests aren't passing 100%, as seen on master, but it will be easier to update them once they live here.

**Comment by willdurand** [comment]
> These tests aren't passing 100%, as seen on master, but it will be easier to update them once they live here.

Can you explain what is failing currently? Also do these tests run on every commit?

**Comment by willdurand** [comment]
I see that there are devhub tests in this patch. I am not sure what the consensus was (I thought all the functional/integration tests were supposed to be moved to a new repo), but I don't think this repo should contain devhub related tests. We won't be able to "fix" them in this repo so we should only have (new) frontend tests.

**Comment by jrbenny35** [comment]
> I see that there are devhub tests in this patch. I am not sure what the consensus was (I thought all the functional/integration tests were supposed to be moved to a new repo), but I don't think this repo should contain devhub related tests. We won't be able to "fix" them in this repo so we should only have (new) frontend tests.

Good point, I'll remove those.

**Comment by jrbenny35** [comment]
> > These tests aren't passing 100%, as seen on master, but it will be easier to update them once they live here.
> 
> Can you explain what is failing currently? Also do these tests run on every commit?

The frontend must haave changed some api calls, as the addons that I create now don't populate the homepage fully. Popular Extensions, and Recommended Extensions are blank, and Recommended Themes isn't displaying. These will need to be updated on -server as the script to setup the site for contributors/local dev lives there.

**Comment by willdurand** [comment]
> These tests aren't passing 100%, as seen on master, but it will be easier to update them once they live here.

Could you please run these tests in this patch for now? I'd like to see what is failing here so that we can take actions and then merge this patch once it is all green.

Also, if it does not take too much time (like not more than right now),  we could maybe run these tests all the time, but I am not sure.

**Comment by jrbenny35** [comment]
> > These tests aren't passing 100%, as seen on master, but it will be easier to update them once they live here.
> 
> Could you please run these tests in this patch for now? I'd like to see what is failing here so that we can take actions and then merge this patch once it is all green.
> 
> Also, if it does not take too much time (like not more than right now), we could maybe run these tests all the time, but I am not sure.

The tests take about 23 min right now. I was hoping I could just move the tests and then update them to green once they are moved, since we are currently running the same suite at the moment.

**Comment by willdurand** [comment]
> The tests take about 23 min right now. I was hoping I could just move the tests and then update them to green once they are moved, since we are currently running the same suite at the moment.

Okay so we will not run them on every commit, that's too long. That being said, it'd be nice to merge these tests once they are all green.

**Comment by jrbenny35** [comment]
@bobsilverberg could you give me some context on what could've changed to cause: 

> Popular Extensions, and Recommended Extensions are blank, and Recommended Themes isn't displaying.

**Comment by jrbenny35** [comment]
Some of the failures are in this report. You can see the screen shots of the page too. https://18631-52104368-gh.circle-artifacts.com/0/home/circleci/addons-frontend/addons-server/ui-test.html

**Comment by bobsilverberg** [comment]
> @bobsilverberg could you give me some context on what could've changed to cause:
> 
> > Popular Extensions, and Recommended Extensions are blank, and Recommended Themes isn't displaying.

I'm not really sure. It looks like nothing is loading into any of the shelves, so it sounds like a missing data issue. Can you point to a commit when these tests started failing?

**Comment by jrbenny35** [comment]
> > @bobsilverberg could you give me some context on what could've changed to cause:
> > > Popular Extensions, and Recommended Extensions are blank, and Recommended Themes isn't displaying.
> 
> I'm not really sure. It looks like nothing is loading into any of the shelves, so it sounds like a missing data issue. Can you point to a commit when these tests started failing?

They have been failing for a little while now. Is there maybe a PR with recent changes?

**Comment by bobsilverberg** [comment]
> > > @bobsilverberg could you give me some context on what could've changed to cause:
> > > > Popular Extensions, and Recommended Extensions are blank, and Recommended Themes isn't displaying.
> > 
> > 
> > I'm not really sure. It looks like nothing is loading into any of the shelves, so it sounds like a missing data issue. Can you point to a commit when these tests started failing?
> 
> They have been failing for a little while now. Is there maybe a PR with recent changes?

I just looked back through 4 weeks of closed PRs (and also the commit history), and don't see anything that touched the home page or would have caused this. Are these the tests that are being moved into this repo? If so, does that mean we don't have a history of them *ever* passing? 

It looks like the console messages from the browser are available via the "Driver Log" link, so I'll take a look at those to see if they provide a clue as to what's happening.

**Comment by jrbenny35** [comment]
Going back through circleci, there were a few problems that were causing failures but nothing to pinpoint an API change.

**Comment by bobsilverberg** [comment]
I'm trying to get what info I can from the Driver Log. It's a bit confusing as I see some of the expected console messages, but others I do not see. The very last console message that indicates an action being dispatched is for `SET_VIEW_CONTEXT` and when I look at the contents of `state` at that point, this is what I see for `shelves`:

```
  shelves: {
    trendingExtensions: null,
    recommendedThemes: [],
    recommendedExtensions: [],
    popularExtensions: [],
    popularThemes: [
      {
        authors: [
          {
            id: 11120,
            name: 'uitest',
            url: 'http://olympia.test/en-US/firefox/user/11120/',
            username: 'uitest',
          },
        ],
        average_daily_users: 4242,
        categories: { firefox: ['holiday'] },
        contributions_url: '',
        created: '2011-02-09T14:01:06Z',
        default_locale: 'en-US',
        description: 'My UI Theme description',
        edit_url: 'http://olympia.test/en-US/developers/addon/ui-test/edit',
        guid: '{5391abbb-cd0f-4642-8038-44d0e46c7ebd}',
        has_eula: false,
        has_privacy_policy: false,
        homepage:
          'https://outgoing.prod.mozaws.net/v1/92587ed9ecb1baa47d6cf9842238ae27b18ccb0fc7d83e23b857ca83c2f70000/https%3A//www.example.org/',
        icon_url: 'http://olympia.test/static/img/addon-icons/default-64.png',
        id: 42,
        is_disabled: false,
        is_experimental: false,
        is_recommended: false,
        last_updated: '2011-02-09T14:01:06Z',
        name: 'Ui-Test Theme',
        previews: [],
        public_stats: true,
        ratings: { average: 5, bayesian_average: 0, count: 777, text_count: 0 },
        requires_payment: false,
        review_url: 'http://olympia.test/en-US/reviewers/review/42',
        slug: 'ui-test',
        status: 'public',
        summary: 'My UI theme summary',
        support_email: 'support@example.org',
        support_url:
          'https://outgoing.prod.mozaws.net/v1/d7a8082e8b86658e28b6f66a666919afd1683982b0a1dc9fa92b0683e6854ad2/https%3A//support.example.org/support/ui-theme-addon/',
        tags: ['another_tag', 'python', 'selenium', 'some_tag', 'ui-testing'],
        type: 'statictheme',
        url: 'http://olympia.test/en-US/firefox/addon/ui-test/',
        weekly_downloads: 123456,
        currentVersionId: 42,
        isRestartRequired: false,
        isWebExtension: true,
        isMozillaSignedExtension: false,
      },
      {
        authors: [
          {
            id: 10969,
            name: 'Firefox user 10969',
            url: 'http://olympia.test/en-US/firefox/user/10969/',
            username: 'nobodymozillaorg',
          },
        ],
        average_daily_users: 1837,
        categories: { firefox: ['nature'] },
        contributions_url: '',
        created: '2019-12-09T17:48:45Z',
        default_locale: 'en-US',
        description: null,
        edit_url:
          'http://olympia.test/en-US/developers/addon/delicious-bacon/edit',
        guid: '@delicious-bacon',
        has_eula: false,
        has_privacy_policy: false,
        homepage: null,
        icon_url: 'http://olympia.test/static/img/addon-icons/photos-64.png',
        id: 29,
        is_disabled: false,
        is_experimental: false,
        is_recommended: false,
        last_updated: '2019-12-09T17:48:45Z',
        name: 'Delicious Bacon',
        previews: [],
        public_stats: false,
        ratings: {
          average: 4,
          bayesian_average: 4.722964604582368,
          count: 5,
          text_count: 5,
        },
        requires_payment: false,
        review_url: 'http://olympia.test/en-US/reviewers/review/29',
        slug: 'delicious-bacon',
        status: 'public',
        summary: null,
        support_email: null,
        support_url: null,
        tags: [],
        type: 'statictheme',
        url: 'http://olympia.test/en-US/firefox/addon/delicious-bacon/',
        weekly_downloads: 208,
        currentVersionId: 29,
        isRestartRequired: false,
        isWebExtension: false,
        isMozillaSignedExtension: false,
      },
      {
        authors: [],
        average_daily_users: 1790,
        categories: { firefox: ['abstract'] },
        contributions_url: '',
        created: '2011-10-05T00:51:01Z',
        default_locale: 'en-US',
        description: null,
        edit_url:
          'http://olympia.test/en-US/developers/addon/add%C3%B4n-ae38933d142147d4ac0d63b3/edit',
        guid: '{a205ee58-169e-4341-a18b-bf4a14c0dc6c}',
        has_eula: false,
        has_privacy_policy: false,
        homepage: null,
        icon_url: 'http://olympia.test/static/img/addon-icons/default-64.png',
        id: 49,
        is_disabled: false,
        is_experimental: false,
        is_recommended: false,
        last_updated: '2011-10-05T00:51:01Z',
        name: 'Add\xF4n ae38933d142147d4ac0d63b3107c3f10',
        previews: [],
        public_stats: false,
        ratings: { average: 0, bayesian_average: 0, count: 0, text_count: 0 },
        requires_payment: false,
        review_url: 'http://olympia.test/en-US/reviewers/review/49',
        slug: 'add\xF4n-ae38933d142147d4ac0d63b3',
        status: 'public',
        summary: 'Summary for Add\xF4n ae38933d142147d4ac0d63b3107c3f10',
        support_email: null,
        support_url: null,
        tags: [],
        type: 'statictheme',
        url:
          'http://olympia.test/en-US/firefox/addon/add%C3%B4n-ae38933d142147d4ac0d63b3/',
        weekly_downloads: 1883,
        currentVersionId: 49,
        isRestartRequired: false,
        isWebExtension: true,
        isMozillaSignedExtension: false,
      },
    ],
  }
```

This looks like `popularThemes` is getting loaded with data, but for some reason none of `recommendedThemes`, `recommendedExtensions` nor `popularExtensions` is, which suggests that perhaps data exists for the former but not for the latter three. Why this might be, I have no idea, but it's a clue, I suppose.

**Comment by jrbenny35** [comment]
Maybe it has something to do with the user. It seems there are calls being made to nginx with this route: ```accounts/account/4757633/collections/privacy-matters/addons/?lang=en-US&wrap_outgoing_links=true```. Account 4757633 does not exist within the integration-test scope.

**Comment by willdurand** [comment]
> Maybe it has something to do with the user. It seems there are calls being made to nginx with this route: `accounts/account/4757633/collections/privacy-matters/addons/?lang=en-US&wrap_outgoing_links=true`. Account 4757633 does not exist within the integration-test scope.

Oh, this user is the special `mozilla` user. We display collections from this user on the homepage.

**Comment by jrbenny35** [comment]
> > Maybe it has something to do with the user. It seems there are calls being made to nginx with this route: `accounts/account/4757633/collections/privacy-matters/addons/?lang=en-US&wrap_outgoing_links=true`. Account 4757633 does not exist within the integration-test scope.
> 
> Oh, this user is the special `mozilla` user. We display collections from this user on the homepage.

I remember @kumar303 telling me of a way I can change that with the node configuration file but I can't seem to find it...

**Comment by jrbenny35** [comment]
Gonna try and change this https://github.com/mozilla/addons-frontend/blob/d428094acb4f9162412f5ac43639842498bea9fd/config/default.js#L362

**Comment by bobsilverberg** [comment]
> > > Maybe it has something to do with the user. It seems there are calls being made to nginx with this route: `accounts/account/4757633/collections/privacy-matters/addons/?lang=en-US&wrap_outgoing_links=true`. Account 4757633 does not exist within the integration-test scope.
> > 
> > 
> > Oh, this user is the special `mozilla` user. We display collections from this user on the homepage.
> 
> I remember @kumar303 telling me of a way I can change that with the node configuration file but I can't seem to find it...

It is in the `config` as `mozillaUserId`. We have a different value for this in the [test config](https://github.com/mozilla/addons-frontend/blob/6fca3e5df1cd88fb293877c929bc433dc7538de7/config/test.js#L20), but it looks like these tests are using the [default value](https://github.com/mozilla/addons-frontend/blob/6fca3e5df1cd88fb293877c929bc433dc7538de7/config/default.js#L362). Perhaps you can control this via the [uitests-amo.js](https://github.com/mozilla/addons-frontend/blob/6fca3e5df1cd88fb293877c929bc433dc7538de7/config/uitests-amo.js) file?

**Comment by jrbenny35** [comment]
Looks like that helped. Still getting some shelves not loading but I will continue to look into it

**Comment by jrbenny35** [comment]
@willdurand with the changes here: https://github.com/mozilla/addons-server/pull/13130, everything should be good to go.

**Comment by willdurand** [comment]
@jrbenny35 would it be possible to run the tests here once?

**Comment by jrbenny35** [comment]
@willdurand I would have to change the ```circleci.yml```

**Comment by willdurand** [comment]
> @willdurand I would have to change the `circleci.yml`

Yeah, that's fine. We'll revert the commit before merging. I want to make sure everything is ok.

**Comment by willdurand** [comment]
woohoo green build!

File: .circleci/config.yml [hunk 1]
```diff
@@ -150,8 +150,8 @@ jobs:
           name: Run tests
           environment:
             MOZ_HEADLESS: 1
-            PYTEST_ADDOPTS: -k "test_search or test_home" -n 4 --reruns 1
-          command: ./tests/ui/run-functional-tests.sh
+            PYTEST_ADDOPTS: -n 4 --reruns 1
```
Comment (willdurand) [inline .circleci/config.yml]: Could you document these options by adding a comment above this line?

File: tests/frontend/ui/README.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
+# Integration Tests for the [Mozilla Addons Website][amo].
```
Comment (willdurand) [inline tests/frontend/ui/README.md]: Is this file still up-to-date?

File: tests/frontend/ui/test_login.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+def test_login(base_url, selenium, fxa_account):
+    """User can login"""
+    page = Home(selenium, base_url).open()
+    assert not page.logged_in
+    page.login(fxa_account.email, fxa_account.password)
+    assert page.logged_in
+
+
+@pytest.mark.skip(
+    reason='https://bugzilla.mozilla.org/show_bug.cgi?id=1453779')
```
Comment (willdurand) [inline tests/frontend/ui/test_login.py]: Is this still a problem?

File: tests/frontend/ui/README.md (hunk 2 above)
Comment (jrbenny35) [reply tests/frontend/ui/README.md]: I'll have to go through it, I suspect it is not.

Comment (willdurand) [reply tests/frontend/ui/README.md]: If not, then I'd suggest we either remove the file or replace its content with only a few words.

File: tests/frontend/ui/test_login.py (hunk 3 above)
Comment (jrbenny35) [reply tests/frontend/ui/test_login.py]: Yes, that bug is still open.

**Pull request merged**